js/web/lib/wasm/jsep/webgpu/ops/attention.ts

```diff
@@ -652,10 +652,25 @@ export const applyAttention = (
   parameters: AttentionParameters,
   attributes: AttentionAttrs,
 ) => {
-  const outputCount = context.outputCount;
-  const pastSequenceLength = parameters.kvNumHeads !== undefined || outputCount > 1 ? parameters.pastSequenceLength : 0;
+  const pastSequenceLength =
+    parameters.kvNumHeads !== undefined || context.outputCount > 1 ? parameters.pastSequenceLength : 0;
   const totalSequenceLength = pastSequenceLength + parameters.kvSequenceLength;
 
+  //
+  // context.outputCount comes from KernelOp and is the number of outputs the op has.
+  // If they are not consumed we need to make sure the shaders don't generate the output
+  // since there is no buffer for it.
+  // We check by requesting the output and if not there we'll adjust context.outputCount
+  //
+  if (parameters.kvNumHeads === undefined) {
```

Comment:
I think kvNumHeads, == undefined is used in this file to distinguish Attention/MultiHeadAttention from GroupQueryAttention. However, that will confuse readers.

I suggest to change the logic of kvNumHeads always assigned a value to indicate the number of heads for key and value. For Attention and MultiHeadAttention operator, it can be assigned the value from num_heads attribute.

The logic here can be changed like the following to fit those 3 operators:
if (context.outputCount > 1) {
       const presentKeyShape = [parameters.batchSize, parameters.kvNumHeads, totalSequenceLength, parameters.headSize];
      ...
}

Reply:
I copied that code from createAttentionProbsProgramInfo().
I can change it here but we might need to do some cleanup in other places.

Reply:
changed.